testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -138,6 +150,11 @@ impl EventDispatcher {
         let mut events: Vec<(Txid, &StacksTransactionEvent)> = vec![];
         let mut i: usize = 0;
         for receipt in chain_tip.receipts.iter() {
+            if receipt.post_condition_aborted {
+                // don't include events for transactions that
+                //   were aborted by post_conditions in the block's events
+                continue;
```

Comment:
Wait, don't we _want_ to report these transaction failures to e.g. the explorer?

Reply:
Yes, they are reported via the transactions field in the dispatched payload, but I think any events generated by those aborted transactions should not be included in the block events (e.g., asset transfers, print statements).

Reply:
As a smart contract developer, I'd want the events, even if the transaction failed. Events are incredibly useful for understanding the execution paths.

Reply:
They are only left out if the post-condition fails -- I assume this would be _very_ confusing to a sidecar consumer, because the events wouldn't correspond to things that actually happened.

Reply:
Ah, I see, missed that.  I'm more aligned with @lgalabru's solution on this, because seeing the event log for a transaction whose post-conditions failed can help me (the developer) figure out why someone else's transaction (e.g. the user) failed to run.  For example, maybe there's a bug in my smart contract that the user's post-conditions caught?  Seeing the event log would help me understand the sequence of steps taken that lead to the bug.

Reply:
We'd want to tag the events from the failed transaction appropriately, so the side-car and upstream consumers know that they do not correspond to materialized state.

Reply:
Fine by me -- I'll add a `committed` boolean to the event JSON objects.